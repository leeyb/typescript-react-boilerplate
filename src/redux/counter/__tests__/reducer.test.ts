import { createAction } from "redux-actions";

import { actionCreators, reducer } from "../index";

const INIT_ACTION = createAction<number>("@@counter/init")(0);
const initialState = reducer(undefined as any, INIT_ACTION);

describe("counter logic", () =>
{
    test("initial state", () =>
    {
        expect(initialState).toMatchSnapshot();
    });

    test("reducer: increased", () =>
    {
        const action = actionCreators.increase(1);
        const state = reducer(initialState, action);
        expect(state.value).toEqual(1);
    });
});
