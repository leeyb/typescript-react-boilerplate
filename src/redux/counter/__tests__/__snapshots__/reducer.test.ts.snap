// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter logic initial state 1`] = `
Object {
  "value": 0,
}
`;
